src/storage/btree.c

```diff
@@ -16226,13 +16412,35 @@ btree_apply_key_range_and_filter (THREAD_ENTRY * thread_p, BTREE_SCAN * bts, boo
   bts->key_range_max_value_equal = false;	/* init as false */
 
   /* Key Range Checking */
+  check_validate (bts);
+
   if (bts->key_range.upper_key == NULL)
     {
-      c = DB_GT;
+      *is_key_range_satisfied = true;
     }
   else
     {
-      c = btree_compare_key (bts->key_range.upper_key, &bts->cur_key, bts->btid_int.key_type, 1, 1, NULL);
+#if defined(IMPROVE_RANGE_SCAN_IN_BTREE)
+      if (bts->is_cur_key_compressed)
+	{
+	  int start_col = 0;
+	  if (start_col < bts->common_prefix_size)
```

Comment:
Is this wrong?
```suggestion
	  if (start_col <= bts->common_prefix_size)
```

Reply:
Even if bts->common_prefix_size=0, it means that there is no information in common_prefix_key.
Thank you.

Reply:
I thought wrong.